src/server/saved_objects/service/lib/repository.js

```diff
@@ -570,13 +586,87 @@ export class SavedObjectsRepository {
       id,
       type,
       updated_at: time,
+      references: response.get._source.references,
       version: response._version,
       attributes: response.get._source[type],
     };
 
 
   }
 
+  /**
+   * Finds other saved objects that have a relationship with a given saved object
+   *
+   * @param {string} type
+   * @param {string} id
+   * @param {object} [options={}]
+   * @property {number} [options.size=10000]
+   * @property {string} [options.namespace]
+   * @property {array} [options.filterTypes]
+   */
+  async findRelationships(type, id, options = {}) {
+    const {
+      size = 10000,
+      namespace,
+      filterTypes = Object.keys(getRootPropertiesObjects(this._mappings)),
+    } = options;
+
+    if (!id || typeof id !== 'string') {
+      throw new TypeError('id must be a string');
+    }
+
+    if (!type || typeof type !== 'string') {
+      throw new TypeError('type must be a string');
+    }
+
+    const { references = [] } = await this.get(type, id, { namespace });
+    const bulkGetOpts = references
+      .filter(ref => filterTypes.includes(ref.type))
+      .map(ref => ({ id: ref.id, type: ref.type }));
+    const searchOpts = {
+      index: this._index,
+      size,
+      from: 0,
+      _source: [
+        'type',
+        'namespace',
+        ...(filterTypes.map(type => `${type}.title`)),
+      ],
+      ignore: [404],
+      rest_total_hits_as_int: true,
+      body: {
+        version: true,
+        query: getRelationshipsQuery({ type, id, namespace, filterTypes }),
+      },
+    };
+
+    const [{ saved_objects: referencedObjects }, referencedResponse, legacyResponse] = await Promise.all([
+      this.bulkGet(bulkGetOpts),
+      this._callCluster('search', searchOpts),
+      // This code will be removed as these types migrate to use "references"
+      legacyFindRelationships({ type, size, namespace, id, savedObjectsClient: this }),
+    ]);
+
+    const relationshipObjects = [].concat(
+      referencedObjects.map(obj => ({ id: obj.id, type: obj.type, ...obj.attributes })),
```

Comment:
By design API should only return `title` from the attributes. Also tests will fail since other unnecessary attributes will be returned.
```suggestion
      referencedObjects.map(obj => ({ id: obj.id, type: obj.type, title: obj.attributes.title })),
```